src/App.tsx

```diff
@@ -1,39 +1,40 @@
 import React from 'react';
 import './App.scss';
-
-function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
-
-  return `Clock-${value}`;
-}
-
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
-
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+import { Clock } from './components/Clock';
+
+export class App extends React.Component {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('contextmenu', (event: MouseEvent) => {
+      event.preventDefault();
+
+      this.setState({ hasClock: false });
+    });
+  }
+
+  componentDidUpdate(): void {
+    if (!this.state.hasClock) {
+      document.addEventListener('click', () => {
+        this.setState({ hasClock: true });
+      });
+    }
+  }
```

Comment:
no need to add a new listener on each update. Do it once in the  componentDidMount

Reply:
and remove listeners in the  `componentWillUnmount`